common/browser.go

```diff
@@ -224,8 +224,27 @@ func (b *Browser) initEvents() error {
 
 // connectionOnAttachedToTarget is called when Connection receives an attachedToTarget
 // event. Returning false will stop the event from being processed by the connection.
-func (b *Browser) connectionOnAttachedToTarget(_ *target.EventAttachedToTarget) bool {
-	return true
+func (b *Browser) connectionOnAttachedToTarget(eva *target.EventAttachedToTarget) bool {
+	// Allow connection to attach to target as the browser context is not set
+	// (hence it's the first one we have—the default), or the target is in the
+	// same browser context as the current one.
+	//
+	// This allows to attach targets to the same browser context as the current
+	// one, and to the default browser context.
+	//
+	// We don't want to hold the lock for the entire function to prevent
+	// concurrency issues with the browser context being closed while we're
+	// checking it. So, we don't use defer. This is for a future code that might
+	// extend this function, add a defer, and cause this issue to be missed.
+	b.contextMu.RLock()
+	defaultContext := b.context == nil
+	if defaultContext || b.context.id == eva.TargetInfo.BrowserContextID {
```

Comment:
Since it's in a short context, I don't think that matters a lot. But I can rename it here.

Reply:
I see your point, but I'm on the side of renaming it because it will help remove ambiguity quickly when doing a find all for `defaultContext` in the code base.

Reply:
Or remove the local `defaultContext` and just add `b.context == nil` directly in the if statement.

Reply:
See b8add515.